preprocessing/shift_generation.py

```diff
@@ -184,29 +205,30 @@ def get_shifts_overlapping_t(shifts, time_sets):
     return shifts_overlapping_t
 
 
-def get_off_shifts(root):
+def get_off_shifts(shifts_per_week):
 
-    events = get_start_events(root)
-
-    off_shifts = []
+    off_shifts = tuplelist()
     off_shifts_in_week = tupledict()
-    week = 0
-    off_shifts_in_week[week] = []
-
-    for i in range(len(events)):
-        for event in events[i:]:
-            duration = event - events[i]
-            if events[i] >= (week + 1) * 24 * 7:
-                week += 1
-                off_shifts_in_week[week] = []
-            if event >= (week + 1) * 24 * 7:
-                break
-            if duration > 70:
-                break
-            elif duration >= 36:
-                if (events[i], duration) not in off_shifts:
-                    off_shifts_in_week[week].append((events[i], duration))
-                    off_shifts.append((events[i], duration))
+
+    for week in shifts_per_week:
+
+        off_shifts_in_week[week] = []
+
+        if shifts_per_week[week][0][0] != (24 * 7 * week):
+            shifts_per_week[week].insert(0, (0.0, 0.0))
```

Comment:
Why are you changing the shifts in a function meant to handle off-shifts?

Reply:
These are dummy-shifts, intending to make it possible to start off-shifts from the first hour of a week.

One skift is added to the start of a week, making an off-shift able to start the first hour.

Also, one shift is added to the end of the week, making an off shift able to end the last hour of a week.

If the final demand period in a week ends on Sunday 18:00, it would otherwise not extract the possible additional 6 hours until 24:00.

Reply:
But shouldn´t these dummy shifts be added to `off_shifts_per_week` and not `shifts_per_week`?

Reply:
They are not off_shifts, and should therefore not be added to any off-shift set. When creating off shifts, the shift-set is used. Thus, the dummy shifts are added to the shift set, so that the off-shift set could be based on it.